packages/react-core-auth/src/hooks/useIsSignedIn/useIsSignedIn.ts

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+import { Hub, HubCallback } from '@aws-amplify/core';
+import { isFunction } from '@aws-amplify/ui';
+
+export interface UseIsSignedInParams {
+  onSignIn?: () => void;
+  onSignOut?: () => void;
+}
+
+interface UseIsSignedIn {
+  isSignedIn: boolean;
+}
+
+const INITIAL_STATE: UseIsSignedIn = { isSignedIn: false };
+
+/**
+ * listens for `Auth` sign in and sign out events
+ *
+ * @param {UseIsSignedInParams} params `onSignIn` and `onSignOut` event callbacks
+ * @returns {UseIsSignedIn} Outputs `isSignedIn`
+ */
+export default function useIsSignedIn({
+  onSignIn,
+  onSignOut,
+}: UseIsSignedInParams): UseIsSignedIn {
+  const [output, setOutput] = React.useState<UseIsSignedIn>(
+    () => INITIAL_STATE
+  );
+
+  const handleEvent: HubCallback = React.useCallback(
+    ({ payload }) => {
+      switch (payload.event) {
+        case 'signIn':
```

Comment:
Are these still going to be the events with V6? I remember seeing this typing when looking at hub events for geo - https://github.com/aws-amplify/amplify-js/blob/next/main/packages/core/src/Hub/types/AuthTypes.ts